### PR TITLE
Fast Track Training Project

### PR DESCRIPTION
**Pull request by Rajith90**
Sort advanced search results based on created date, author, and average rating
Sort activity search results based on action performed date.

### CODE REVIEW

File: components/registry/org.wso2.carbon.registry.activities.ui/src/main/resources/web/activities/activity-ajaxprocessor.jsp [hunk 1]
```diff
@@ -41,6 +41,19 @@
         ActivityBean activityBean;
         String requestedPage = request.getParameter(UIConstants.REQUESTED_PAGE);
 
+        String sortOrder = request.getParameter("sortOrder");
+        String imgType = "";
+        String displayStr = "display:none;";
+
+        if (sortOrder!=null){
+            if (sortOrder.equals("DES")){
```
Comment (sameerak) [inline components/registry/org.wso2.carbon.registry.activities.ui/src/main/resources/web/activities/activity-ajaxprocessor.jsp]: sortOrder.equals("DES") can be replaced by,
"DES".equals(sortOrder) and eliminate the usage of null check in line 48.

Same solution can be used for below line 51.
After that re-think weather that null check is necessary for line 54 code.

File: components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js [hunk 2]
```diff
@@ -226,10 +226,10 @@ function initMiscFields() {
     }
 }
 
-function submitAdvSearchForm(pageNumber) {		
+function submitAdvSearchForm(pageNumber) {
```
Comment (sameerak) [inline components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js]: remove unnecessary formatting

File: components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js [hunk 3]
```diff
@@ -226,10 +226,10 @@ function initMiscFields() {
...
     sessionAwareFunction(function() {
         document.getElementById('advancedSearchFormDiv').style.display = "none";
-	var resourceName = document.getElementById('#_resourceName');
+	    var resourceName = document.getElementById('#_resourceName');
```
Comment (sameerak) [inline components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js]: remove unnecessary formatting

File: components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js [hunk 4]
```diff
@@ -398,6 +398,104 @@ function submitAdvSearchForm(pageNumber) {
...
+ * Method to send sort request to backend using Ajax call.
+ *
+ * @param pageNumber    current page of the search results
+ * @param sortOrder     ascending or descending
+ * @param sortBy        sort based on created date, author and rating
+ */
+function sort(pageNumber,sortOrder, sortBy) {
+    sessionAwareFunction(function() {
+        document.getElementById('advancedSearchFormDiv').style.display = "none";
+	    var resourceName = document.getElementById('#_resourceName');
```
Comment (sameerak) [inline components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js]: remove unnecessary formatting

File: components/registry/org.wso2.carbon.registry.search/src/main/java/org/wso2/carbon/registry/search/services/SearchService.java [hunk 5]
```diff
@@ -40,11 +40,13 @@
...
 import org.wso2.carbon.registry.search.services.utils.SearchUtils;
 
+import java.util.Arrays;
+import java.util.Comparator;
 import java.util.HashMap;
 import java.util.Map;
 
 /**
- *  This class is dedicated for resource search
+ * This class is dedicated for resource search
```
Comment (sameerak) [inline components/registry/org.wso2.carbon.registry.search/src/main/java/org/wso2/carbon/registry/search/services/SearchService.java]: remove unnecessary formatting

File: components/registry/org.wso2.carbon.registry.activities.ui/src/main/resources/web/activities/activity-ajaxprocessor.jsp (hunk 1 above)
Comment (Rajith90) [inline components/registry/org.wso2.carbon.registry.activities.ui/src/main/resources/web/activities/activity-ajaxprocessor.jsp]: Since initial loading of activity list page does not contain sort order as parameter, and the display string depends on the value of sort order. There for null check is required. Because null check is already done, then no need to replace with
"DES".equals(sortOrder) 

Thanks!

File: components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js (hunk 2 above)
Comment (Rajith90) [inline components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js]: There were unnecessary tab spacing were included.  It was removed by the formatting. Hence the formatting is correct

File: components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js [hunk 6]
```diff
@@ -398,6 +398,115 @@ function submitAdvSearchForm(pageNumber) {
...
+    sessionAwareFunction(function() {
+        document.getElementById('advancedSearchFormDiv').style.display = "none";
+        var resourceName = document.getElementById('#_resourceName');
+        var searchResuts = $('searchResuts');
+        searchResuts.style.display = "";
+        searchResuts.innerHTML = org_wso2_carbon_registry_search_ui_jsi18n["searching"];
+
+        var table = $('customTable');
+        var rows = table.getElementsByTagName('input');
+
```
Comment (prasa7) [inline components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js]: Unnecessary white spaces. Check the other places as wel

File: components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js [hunk 7]
```diff
@@ -398,6 +398,115 @@ function submitAdvSearchForm(pageNumber) {
...
+            {
+                method: 'get',
+                parameters: {parameterList: customParamterList, sortOrder: sortOrder, sortBy: sortBy},
+                evalScripts: true
+            });
+        }
+        $('#_0').focus();
+
+    }
+        ,
```
Comment (prasa7) [inline components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js]: I think there is a formatting issue

Comment (Rajith90) [inline components/registry/org.wso2.carbon.registry.search.ui/src/main/resources/web/search/js/search.js]: Fixed in the same PR

**Comment by daneshk** [comment]
Closing this PR, since this implementation is no longer valid with new search api changes.
